designs/usb-light.stanza

```diff
@@ -31,15 +43,75 @@ pcb-module main-board:
+    val net-id1 = make-net( to-symbol(net-name1), [(led[i + 4].a)])
+    net (net-id1 led[i].c)
+
+    val net-name2 = string-join(["net-res", i + 1])
+    val net-id2 = make-net( to-symbol(net-name2), [(res[i].p[1])])
+    net (net-id2 led[i + 4].c)
+
     net (gnd res[i].p[2])
-    place(led[i + 4]) at loc(-2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(led[i]) at loc(-7.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(res[i]) at loc(2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-  place(conn) at loc(10.75, 0.0, 90.0) on Top
+    ; place p5v track stub
+    val led-5v-x1 = x(led-i-point) - y(center(get-pad-pose(((led[i])), "a")))
+    println("%_ %_ %_ " % [led-5v-x1 x(led-i-point)  y(center(get-pad-pose(((led[i])), "a"))) ])
+    val led-5v-y1 = y(led-i-point) 
+    geom(p5v) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(led-5v-x1, led-5v-y1), Point(led-5v-x1 - 2.0, led-5v-y1) ])
+    ; place p5v track 
+      if i != 0 :
+        copper(LayerIndex(0)) = Line(trace-thickness, [Point(led-5v-x1 - 2.0, led-5v-y1) Point(led-5v-x1 - 2.0, led-5v-y1 - 5.0) ])      
+      else :
+        copper(LayerIndex(0)) = Polyline(trace-thickness, [Arc(Point(led-5v-x1 - 2.0, led-5v-y1), Point(-9.5, -9.5), Point(-8.0, -10.0))])
+
+    ; place track between the LEDs
+    val led-5v-x2 = x(led-i-point) - y(center(get-pad-pose(((led[i])), "c")))
+    val led-middle-x1 = x(led-i4-point) - y(center(get-pad-pose(((led[i + 4])), "a")))
+    println("led-5v-x2 %_" % [led-5v-x2])
+    geom(net-id1) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(led-5v-x2, led-5v-y1), Point(led-middle-x1, led-5v-y1) ])
+
+    ; place track between LED and resistor
+    val led-middle-x2 = x(led-i4-point) - y(center(get-pad-pose(((led[i + 4])), "c")))
+    val res-i-x1 = x(res-i-point) - y(center(get-pad-pose(((res[i])), "p",1)))
+    geom(net-id2) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(res-i-x1, led-5v-y1), Point(led-middle-x2, led-5v-y1) ])
+
+    ; place ground track stub
+    val res-i-x2 = x(res-i-point) - y(center(get-pad-pose(((res[i])), "p",2)))
+    geom(gnd) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(res-i-x2, led-5v-y1), Point(5.0, led-5v-y1) ])
+      if i != 0 :
+        copper(LayerIndex(0)) = Line(trace-thickness, [Point(5.0, led-5v-y1) Point(5.0, led-5v-y1 - 5.0) ])
+  val conn-point = Point(10.75, 0.0)
+  place(conn) at loc(conn-point, 90.0) on Top
+  val conn-pin-x = x(conn-point) - y(center(get-pad-pose(((conn)), "p",1)))
+  val conn-gnd-y = y(conn-point) + x(center(get-pad-pose(((conn)), "p",5)))
+  val conn-p5v-y = y(conn-point) + x(center(get-pad-pose(((conn)), "p",1)))
+  geom(p5v) :
+    copper(LayerIndex(0)) = Line(trace-thickness, [ Point(5.7, conn-p5v-y), Point(conn-pin-x, conn-p5v-y) ])
+    copper(LayerIndex(0)) = Line(trace-thickness, [ Point(5.7, -8.0), Point(5.7, conn-p5v-y) ])
+    ; copper(LayerIndex(0)) = Line(trace-thickness, [ Point(5.7, -10.0), Point(-8.0, -10.0) ])
```

Comment:
I'd remove this or explain why it's here
```suggestion
```

Reply:
removed